src/components/community-link/community-link.js

```diff
@@ -0,0 +1,52 @@
+import '@brightspace-ui/core/components/link/link.js';
+
+import { getDocumentLocaleSettings } from '@brightspace-ui/intl/lib/common.js';
+
+const Link = customElements.get('d2l-link');
```

Comment:
Hmm, interesting! I think I'd rather either:

1. We add an `export` to link's file for the class so you can just `import { LinkClass } from '@brightspace-ui/core/components/link/link.js';`. More realistically we should probably split it into two files -- one that defines the class and one that imports it and registers the custom element. That way you could import it without registering.
2. You just expose whatever of link's properties you want here, import `linkStyles` and apply them to your own `<a class="d2l-link">` element.

If you don't need all the crazy link properties, I'd probably lean towards 2. It's already a bit weird that the consumer of this component can set the `href` and then it'll get blown away.

It would also give you the opportunity to reign in what consumers can do if you like. Like do you want them to be able to put _anything_ inside the link? If not, you could add a `text` property and just always render that. Will the links _always_ open in a new window? Then don't even let them set `target` and just always apply that yourself. And so on.

Reply:
Yeah my thinking here is that it would be a true "inheritor" of d2l-link. My first implementation was just taking and passing all the parent community link properties into a child link which seemed silly.

> It's already a bit weird that the consumer of this component can set the href and then it'll get blown away.

I could remove href as a property?

> Will the links always open in a new window?

For our use case yes, but I don't know about for all uses. Using the component to set a standard (all community links open in a new tab) maybe is desired?

>  Like do you want them to be able to put anything inside the link?

Could be a button? For insights dashboard community links they are a help button. Being able to slot in a button would be pretty handy

Reply:
Semantically it should always be a link and not a button, since links can be opened in new tabs, bookmarked, etc. whereas buttons cannot. _Visually_ it could still look like a button, although we still try really hard to avoid that.

For me this is making more and more of an argument for this not inheriting from `Link` and just let it define its own properties. `<d2l-labs-link-community type="button|link" text="Get Help on Something">`.

Reply:
That a good point... Okay I'm convinced, I'll make those change